locations/spiders/gelsons.py

```diff
@@ -0,0 +1,71 @@
+import json
+import scrapy
+import re
+
+from locations.items import GeojsonPointItem
+from locations.hours import OpeningHours, DAYS
+
+
+class GelsonsSpider(scrapy.Spider):
+    name = "gelsons"
+    item_attributes = {"brand": "Gelson's Markets", "brand_wikidata": "Q16993993"}
+    allowed_domains = ["gelsons.com"]
+    download_delay = 0.2
+    start_urls = ("https://www.gelsons.com/sitemap.xml",)
+
+    def parse(self, response):
+        response.selector.remove_namespaces()
+        urls = response.xpath("//url/loc/text()").extract()
+        for url in urls:
+            if re.match(r".*stores/.+$", url) and "virtual" not in url:
+
+                yield scrapy.Request(
+                    url=url, callback=self.parse_store, meta={"url": url}
+                )
+
+    def parse_store(self, response):
+
+        content = json.loads(
+            response.xpath('//script[@type="application/json"]/text()').extract_first()
+        )["props"]["pageProps"]
+        store_json = content["store"]
+        hours = content["pageComponents"][0]["headline"]
+
```

Comment:
Hi,
item=DictParser.parse(store_json) should do quite a bit of what you have typed in below. Bits it does not do then you can still do! Bit like this:
https://github.com/alltheplaces/alltheplaces/blob/master/locations/spiders/greggs_gb.py
Gives your keyboard a bit more life :-) Thank you for the PR.
Cheers,
Joe.

Reply:
Thanks for the tip!